Lib/ufo2ft/filters/dottedCircleFilter.py

```diff
@@ -1,262 +1,21 @@
-"""
-Dotted Circle Filter
-
-This filter checks whether a font contains a glyph for U+25CC (DOTTED CIRCLE),
-which is inserted by complex shapers to display mark glyphs which have no
-associated base glyph, usually as a result of broken clusters but also for
-pedagogical reasons. (For example, to display the marks in a table of glyphs.)
-
-If no dotted circle glyph is present in the font, then one is drawn and added.
-
-Next, the filter creates any additional anchors for the dotted circle glyph to
-ensure that all marks can be attached to it. It does this by gathering a list
-of anchors, finding the set of base glyphs for each anchor, computing the
-average position of the anchor on the base glyph (relative to the glyph's width),
-and then creating an anchor at that average position on the dotted circle glyph.
-
-The filter must be run as a "pre" filter. This can be done from the command
-line like so::
-
-    fontmake -o ttf -g MyFont.glyphs --filter "DottedCircleFilter(pre=True)"
-
-or in the ``lib.plist`` file of a UFO::
-
-    <key>com.github.googlei18n.ufo2ft.filters</key>
-    <array>
-      <dict>
-        <key>name</key>
-        <string>DottedCircleFilter</string>
-        <key>pre</key>
-        <true/>
-      </dict>
-    </array>
-
-The filter supports the following options:
-
-margin
-    When drawing a dotted circle, the vertical space in units around the dotted
-    circle.
-sidebearings
-    When drawing a dotted circle, additional horizontal space in units around
-    the dotted circle.
-dots
-    Number of dots in the circle.
+"""NOTE: This is a redirection stub, because the original module was misnamed.
 
+It ended in "Filter", which messed with the assumptions in ``getFilterClass``.
+Make an alias for the class here to steer the importer in the right
+direction.
```

Comment:
I don't really like this `DottedCircleFilterFilter` hack.. I think you need to issue a warnings.warn if the deprecated `dottedCircleFilter` module gets imported, so move this warning to the module scope. You don't need a `*FilterFilter` subclass...

Reply:
Just ignore `Filter` suffix in the filter name if present, I think you did that at some point but then reverted (https://github.com/googlefonts/ufo2ft/pull/715)